src/scripts/components/UploadListItem.js

```diff
@@ -0,0 +1,567 @@
+        this.props.checkUserWallet()
+      }
+    }
+  }
+
+  publishVideo (publish: false) {
+    this.props.openModal(MODAL.STAKE)
+  }
+
+  toggleOpen (e: Object) {
+    this.setState({
+      open: !this.state.open
+    })
+  }
+
+  handleHeight (e: Object) {
+    let height = '0px'
+    if (this.formWrapperRef && this.state.open) {
+      height = this.formWrapperRef.offsetHeight + 'px'
+    }
+    return height
+  }
+
+  render () {
+    const { video, videoId } = this.props
+    // Title
+    const title = video.title || video.filename
+
+    // Need to do this below
+    // if (!video || !video.id) {
+    //   return <ListItem>Something went wrong - no video known</ListItem>
+    // }
+
+    // Status
+    let videoIsReady = false
+    let statusMessage = ''
+
+    if (
+      video.uploadStatus.name === 'success' &&
+      video.transcodingStatus.name === 'success'
+    ) {
+      if (video.title.length < 1) {
+        statusMessage = 'Please provide a title and description'
+      } else {
+        statusMessage = 'Your video is ready'
+        videoIsReady = true
+      }
+    } else {
+      if (video.uploadStatus.name === 'failed') {
+        statusMessage = 'Your video could not be uploaded'
+      } else if (video.transcodingStatus.name === 'failed') {
+        statusMessage = 'Your video could not be transcoded'
+      } else if (video.uploadStatus.name === 'success') {
+        statusMessage = 'Transcoding your video'
+      } else {
+        statusMessage = 'Uploading your video'
+      }
+    }
+
+    // Progress
+    const progress = videoProgress(video)
+
+    // Media box
+    let poster = ''
+    let videoPoster = ''
+    let videoMediaBox = null
+    const isPublished = isVideoPublished(video)
+    const isPublishable = isVideoPublishable(video)
+    const duration = videoDuration(video)
+
+    let durationBox = null
+    if (duration) {
+      const durationNoMillis = duration.substring(0, duration.indexOf('.'))
+      durationBox = (
+        <VideoMediaTime>
+          <VideoMediaTimeText>{durationNoMillis}</VideoMediaTimeText>
+        </VideoMediaTime>
+      )
+    }
+
+    const urlToPlay = '/play/' + video.id
+    const urlForSharing =
+      getAppRootUrl(process.env.NODE_ENV) + '/play/' + video.id
+
+    if (video && video.thumbnails.size === 4) {
+      poster = video.thumbnails.get(0)
+      videoPoster =
+        'https://gateway.paratii.video/ipfs/' + video.ipfsHash + '/' + poster
+    }
+
+    if (isPublished) {
+      videoMediaBox = (
+        <VideoMedia>
+          <VideoMediaLink to={urlToPlay}>
+            <VideoMediaOverlay>
+              <VideoMediaIcon>
+                <SVGIcon color="white" icon="icon-player-play" />
+              </VideoMediaIcon>
+              {durationBox}
+            </VideoMediaOverlay>
+            <VideoImage source={videoPoster} />
+          </VideoMediaLink>
+        </VideoMedia>
+      )
+    } else if (isPublishable) {
```

Comment:
Generally a good practice to not add this much logic into `render() {}`. Also, instead of using `let` variables in `render` it is more conventional (and easier to read) if you do this:

```javascript
<div>
  { isPublished ? <something /> : <somethingElse /> }
</div>
```

☝️ but the above should still go in a separate render method or even component